--- amsn/plugins/music/infosongbird	(revision 11807)
+++ amsn/plugins/music/infosongbird	(working copy)
@@ -1,7 +1,10 @@
 #!/usr/bin/env python
-import dbus
-bus = dbus.SessionBus()
-object = bus.get_object('org.mozilla.songbird', '/org/mozilla/songbird')
-print object.getStatus()
-print object.getTitle()
-print object.getArtist()
+try:
+	import dbus
+	bus = dbus.SessionBus()
+	object = bus.get_object('org.mozilla.songbird', '/org/mozilla/songbird')
+	print object.getStatus()
+	print object.getTitle()
+	print object.getArtist()
+except dbus.exceptions.DBusException:
+	print "not installed"
