--- amsn/autoupdate.tcl	2009-11-11 09:23:42.736604374 +0100
+++ amsn~/autoupdate.tcl	2009-11-11 09:38:25.349355120 +0100
@@ -536,7 +536,7 @@
 	}
 
 	#///////////////////////////////////////////////////////////////////////
-	package require -exact http 2.4.4
+	package require http 2.4.4
 
 	proc check_web_version { token } {
 		global version rcversion weburl
--- amsn/proxy.tcl	2009-11-11 09:23:43.270603596 +0100
+++ amsn~/proxy.tcl	2009-11-11 09:31:51.524364201 +0100
@@ -9,7 +9,7 @@
 ::Version::setSubversionId {$Id$}
 
 package provide Proxy 0.1
-package require -exact http 2.4.4
+package require http 2.4.4
 
 # This should be converted to a proper package, to use with package require
 source socks.tcl	;# SOCKS5 proxy support
