--- amsn-0.97.2/amsn.desktop~	2008-07-10 23:54:42.000000000 +0300
+++ amsn-0.97.2/amsn.desktop	2009-05-11 18:54:01.216674578 +0300
@@ -6,9 +6,12 @@
 Comment=MSN Messenger for Linux
 Comment[fr]=MSN Messenger pour Linux
 Comment[de]=MSN Messenger für Linux
+Comment[nb]=MSN Messenger for Linux
 Comment[nl]=MSN Messenger voor Linux
 Comment[no]=MSN Messenger for Linux
 Comment[es]=MSN Messenger para Linux
+Comment[pl]=MSN Messenger dla Linuksa
 Terminal=false
 Type=Application
 StartupNotify=true
+# vi: encoding=utf-8
