--- amsn-0.96.orig/amsn.desktop	2007-03-05 22:43:49.403398500 +0100
+++ amsn-0.96/amsn.desktop	2007-03-05 23:02:37.065873000 +0100
@@ -4,12 +4,14 @@
 Exec=amsn
 Icon=msn.png
 Info=AMSN
-Categories=Application;Network;
+Categories=Network;InstantMessaging;
 Comment=MSN Messenger for Linux
 Comment[fr]=MSN Messenger pour Linux
 Comment[de]=MSN Messenger fr Linux
+Comment[nb]=MSN Messenger for Linux
 Comment[nl]=MSN Messenger voor Linux
 Comment[no]=MSN Messenger for Linux
+Comment[pl]=MSN Messenger dla Linuksa
 Terminal=false
 Type=Application
 StartupNotify=true
