diff -burNp amsn-0.97/amsn.desktop amsn-0.97-new/amsn.desktop
--- amsn-0.97/amsn.desktop	2007-12-06 19:23:54.000000000 +0100
+++ amsn-0.97-new/amsn.desktop	2008-10-26 08:50:12.616669979 +0100
@@ -6,8 +6,12 @@ Categories=Network;InstantMessaging;
 Comment=MSN Messenger for Linux
 Comment[fr]=MSN Messenger pour Linux
 Comment[de]=MSN Messenger für Linux
+Comment[nb]=MSN Messenger for Linux
 Comment[nl]=MSN Messenger voor Linux
 Comment[no]=MSN Messenger for Linux
+Comment[es]=MSN Messenger para Linux
+Comment[pl]=MSN Messenger dla Linuksa
 Terminal=false
 Type=Application
 StartupNotify=true
+# vi: encoding=utf-8
