diff -Nur amsn-0.96.orig/utils/linux/capture/libng/grab-ng.c amsn-0.96/utils/linux/capture/libng/grab-ng.c
--- amsn-0.96.orig/utils/linux/capture/libng/grab-ng.c	2007-03-05 22:43:48.135319250 +0100
+++ amsn-0.96/utils/linux/capture/libng/grab-ng.c	2007-03-05 22:44:31.706042250 +0100
@@ -1144,7 +1144,7 @@
 	    if (NULL == (initcall = dlsym(plugin,"_ng_plugin_init"))) {
 		continue;
 	    }
-	}
+	};
 #if 0
 	initcall();
 #endif
